code/modules/telesci/rcs.dm

```diff
@@ -38,84 +48,79 @@
 	QDEL_NULL(rcell)
 	return ..()
 
+/**
+  * Used to select telepad location.
+  */
 /obj/item/rcs/attack_self(mob/user)
-	if(!emagged)
+	if(teleporting)
+		to_chat(user, "<span class='warning'>Error: Unable to change destination while in use.</span>")
 		return
-	if(mode == RCS_MODE_CALIBRATED)
-		mode = RCS_MODE_UNCALIBRATED
-		playsound(get_turf(src), 'sound/effects/pop.ogg', 50, 0)
-		to_chat(user, "<span class='caution'>The telepad locator has become uncalibrated.</span>")
-	else
-		mode = RCS_MODE_CALIBRATED
-		playsound(get_turf(src), 'sound/effects/pop.ogg', 50, 0)
-		to_chat(user, "<span class='caution'>You calibrate the telepad locator.</span>")
-
-/obj/item/rcs/emag_act(user as mob)
+
+	var/list/L = list() // List of avaliable telepads
+	var/list/areaindex = list() // Telepad area location
+	for(var/obj/machinery/telepad_cargo/R in GLOB.machines)
+		if(R.stage)
+			continue
+		var/turf/T = get_turf(R)
+		var/locname = T.loc.name // The name of the turf area. (e.g. Cargo Bay, Experimentation Lab)
+
+		if(areaindex[locname]) // If there's another telepad with the same area, increment the value so as to not override (e.g. Cargo Bay 2)
+			locname = "[locname] ([++areaindex[locname]])"
+		else // Else, 1
+			areaindex[locname] = 1
+		L[locname] = R
+
+	if(emagged) // Add an 'Unknown' entry at the end if it's emagged
+		L += "**Unknown**"
+
+	var/select = input("Please select a telepad.", "RCS") in L
+	if(select == "**Unknown**") // Randomise the teleport location
+		rand_x = rand(50, 200)
+		rand_y = rand(50, 200)
+		pad = locate(rand_x, rand_y, 6)
+	else // Else choose the value of the selection
+		pad = L[select]
+	playsound(src, 'sound/effects/pop.ogg', 25, TRUE) // And play a sound either way.
+
+
+/obj/item/rcs/emag_act(user)
 	if(!emagged)
 		emagged = TRUE
-		do_sparks(5, 1, src)
-		to_chat(user, "<span class='caution'>You emag the RCS. Activate it to toggle between modes.</span>")
+		do_sparks(3, TRUE, src)
+		to_chat(user, "<span class='boldwarning'>Warning: Safeties disabled.</span>")
 		return
 
+
 /obj/item/rcs/proc/try_send_container(mob/user, obj/structure/closet/C)
 	if(teleporting)
 		to_chat(user, "<span class='warning'>You're already using [src]!</span>")
 		return
-	if(user in C.contents) //to prevent self-teleporting.
+	if((user in C.contents) && (!emagged)) // If it's emagged, skip this check.
+		to_chat(user, "<span class='warning'>Error: User located in container--aborting for safety.</span>")
 		return
-	if(!rcell || (rcell.charge < chargecost))
-		to_chat(user, "<span class='warning'>Out of charges.</span>")
+	if(!rcell || rcell.charge < chargecost)
```

Comment:
Do you think its necessary to check if `rcell` exists? From what I can tell, there's no way to remove a cell from an RCS so I think we can just assume one is there.